daemons/attrd/pacemaker-attrd.c

```diff
@@ -370,16 +370,16 @@ main(int argc, char **argv)
     }
     crm_info("Cluster connection active");
 
+    writer = election_init(T_ATTRD, attrd_cluster->uname, 120000, attrd_election_cb);
+
```

Comment:
What are the consequences?
Faster (up to above 2 minutes of speed-up) synchronization among attrd peers?
More cumulated initial I/O (and possibly other resources spike)
since much more is going on in the initial phase in other daemons?

Reply:
attrd_cib_connect() (now below this) calls attrd_broadcast_protocol(), which calls attrd_client_update() to set the #attrd-protocol attribute. That triggers an election, which previously would immediately fail because the election pointer was NULL. The intent is to avoid the unnecessary failure. The effect is not particularly significant, since an election will be called if any value needs to be written out, but it's the correct order of operation.

It is also a good idea to initialize the election as soon as possible after cluster connection so we are able to process any incoming election messages. (There's room for more improvement there, but it would involve API changes, so I'm not addressing it yet.)